backend/app/routers/aiod.py

```diff
@@ -103,7 +103,7 @@ async def get_my_datasets_count(
 async def create_dataset(
     dataset: Dataset,
     token: str = Depends(get_current_user_token),
-    user: dict = Depends(get_current_user(required=True)),
+    user: dict = Depends(get_current_user(required=True, from_api_key=True)),
```

Comment:
It won't be possible to use API key for this endpoint at the moment since a valid token would be needed on the AIoD side.